src/basic/time-util.c

```diff
@@ -868,12 +836,38 @@ static int parse_timestamp_impl(const char *t, usec_t *ret, bool with_tz) {
         if (weekday >= 0 && tm.tm_wday != weekday)
                 return -EINVAL;
 
+        /* gmtoff may be negative, and the date maye be too old (e.g. 1969-12-31 23:00:00 -06). */
+        if (gmtoff < 0 && tm.tm_year == 69 && tm.tm_mon == 11 && tm.tm_mday == 31) {
```

Comment:
why is this date hardcoded? could it not be tm_year=68 or so? would be good to have a test for this corner case